src/validator.js

```diff
@@ -537,19 +537,19 @@ JSONEditor.Validator = Class.extend({
     // date, time and datetime-local validation
     if(['date', 'time', 'datetime-local'].indexOf(schema.format) != -1) {
 
-      var validator = {
-          'date': /^(\d{4}\D\d{2}\D\d{2})?$/,
-          'time': /^(\d{2}:\d{2}(?::\d{2})?)?$/,
-          'datetime-local': /^(\d{4}\D\d{2}\D\d{2} \d{2}:\d{2}(?::\d{2})?)?$/
+      var validatorRx = {
+        'date': /^(\d{4}\D\d{2}\D\d{2})?$/,
+        'time': /^(\d{2}:\d{2}(?::\d{2})?)?$/,
+        'datetime-local': /^(\d{4}\D\d{2}\D\d{2}[ T]\d{2}:\d{2}(?::\d{2})?)?$/
```

Comment:
Is it on purpose that you allow a ` `(space) in the regex?

Reply:
Yes. Chrome returns the date string differently from Firefox and others. So it has to be able to validate with either space or "T"

Reply:
Do we need to take care about the different browsers regarding reformatting the values?
Would people using FF or Chrome save different values to a database? Or would this have to be done on the server?

Reply:
Storing dates as strings in databases is generally not a good idea. A integer epoch date (unix timestamp) is much better.  
The string format returned is based on the format selected (flatpickr) or the default browser return value.